hostapd: add support for setting sae_pwe

huaracheguarache opened the pull request:
Make it possible to specify the SAE mechanism for PWE derivation. The
following values are possible:

0 = hunting-and-pecking loop only (default without password indentifier)
1 = hash-to-element only (default with password identifier)
2 = both hunting-and-pecking loop and hash-to-element enabled

The default is 0, as this is the current default in hostapd.

Signed-off-by: Michael Yartys <michael.yartys@protonmail.com>

hauke commented:
I would prefer to leave sae_pwe unset if it is not set explicitly, it uses different values if a password identifier is set or not, currently we do not support setting password identifier.

huaracheguarache commented:
Crap, I messed this PR up. I've created a new one (#4200) where a default sae_pwe value is not set, @hauke.

huaracheguarache commented:
@adschm Sorry for double posting, I've fixed this one now.

@hauke Ok, now it doesn't explicitly set a default value, it just follows the hostapd default to use hunting-and-pecking loop only.

adschm commented:
This requires a PKG_RELEASE bump in the Makefile. One could consider switching to AUTORELEASE instead ...

adschm commented:
I just saw that we forgot to decrement the PKG_RELEASE in https://github.com/openwrt/openwrt/commit/eefed841b05c3cd4c65a78b50ce0934d879e6acf , so AUTORELEASE won't work until the next update as it would decrement the number now. But may if somebody picks the patch that person can also bump package release manually for now.

huaracheguarache commented:
It looks like it would have worked though. This was the resulting package names after I compiled a build prior to your package bump:

```
hostapd-common_2021-05-21-d98f4c46-6_arm_cortex-a15_neon-vfpv4.ipk
hostapd-openssl_2021-05-21-d98f4c46-6_arm_cortex-a15_neon-vfpv4.ipk
```

The -6 suffix counts 5 commits after the package bump, and that includes my $(AUTORELEASE) commit.

I'll get rid of the $(AUTORELEASE) commit and push again.

huaracheguarache commented:
Anything else that needs to be done, or does this look good to you, @adschm?

huaracheguarache committed:
hostapd: add support for setting sae_pwe

Make it possible to specify the SAE mechanism for PWE derivation. The
following values are possible:

0 = hunting-and-pecking loop only
1 = hash-to-element only
2 = both hunting-and-pecking loop and hash-to-element enabled

hostapd currently defaults to hunting-and-pecking loop only.

Signed-off-by: Michael Yartys <michael.yartys@protonmail.com>

huaracheguarache commented:
Updated to make patch apply after the recent changes to hostapd, starting with: https://git.openwrt.org/?p=openwrt/openwrt.git;a=commit;h=46509a51dd63aa49648ad0f130461817f43532d0

huaracheguarache commented:
Just bumping @hauke. I've tested with today's master and everything still works. Is there anything else I need to do, or is this ready to be merged?

aparcar commented:
Since the AUTORELEASE issue was resolved and the patch looks good I merged this. Thank you!

Neustradamus commented:
Merged commit is:
- https://github.com/openwrt/openwrt/commit/b9b4aef4f2403766a1a3321b0ef095c2728a3139
